### PR TITLE
NUTCH 2940 Develop Gradle Core Build for Apache Nutch

### PR DESCRIPTION
**Pull request by jbsimmon**
Added file collection variable for test classpath, and gradle tasks for ant targets compile, compile-core, proxy, benchmark. Also fixed a depends for previously renamed init-nutch.

Thanks for your contribution to [Apache Nutch](https://nutch.apache.org/)! Your help is appreciated!

Before opening the pull request, please verify that
* there is an open issue on the [Nutch issue tracker](https://issues.apache.org/jira/projects/NUTCH) which describes the problem or the improvement. We cannot accept pull requests without an issue because the change wouldn't be listed in the release notes.
* the issue ID (`NUTCH-XXXX`)
  - is referenced in the title of the pull request
  - and placed in front of your commit messages surrounded by square brackets (`[NUTCH-XXXX] Issue or pull request title`)
* commits are squashed into a single one (or few commits for larger changes)
* Java source code follows [Nutch Eclipse Code Formatting rules](https://github.com/apache/nutch/blob/master/eclipse-codeformat.xml)
* Nutch is successfully built and unit tests pass by running `ant clean runtime test`
* there should be no conflicts when merging the pull request branch into the *recent* master branch. If there are conflicts, please try to rebase the pull request branch on top of a freshly pulled master branch.

We will be able to faster integrate your pull request if these conditions are met. If you have any questions how to fix your problem or about using Nutch in general, please sign up for the [Nutch mailing list](https://nutch.apache.org/mailing_lists.html). Thanks!

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -73,7 +73,19 @@ val classpathCollection: FileCollection = layout.files(
...
 )
-val classpath: String = classpathCollection.asPath
+val classPath: String = classpathCollection.asPath
+
+// test classpath
+val testClasspathCollection: FileCollection = layout.files(
+    file(project.properties["test.build.classes"]),
+    file(project.properties["conf.dir"]),
+    file(project.properties["test.src.dir"]),
+    file(project.properties["build.plugins"]), // plugins.classpath.dir
```
Comment (lewismc) [inline build.gradle.kts]: Please remove comment

File: build.gradle.kts [hunk 2]
```diff
@@ -96,6 +108,69 @@ tasks.register<Copy>("init-nutch") {
...
+    
+    options.annotationProcessorPath = classpathCollection
+    options.sourcepath = layout.files("${project.properties["src.dir"]}")
+    options.compilerArgs.add("-Xlint-path")
+    options.encoding = "${project.properties["build.encoding"]}"
+    options.isDebug = "${project.properties["javac.debug"]}" == "on"
+    options.isDeprecation = "${project.properties["javac.deprecation"]}" == "on"
+
+    // Deprecated?
+    // options.isOptimize = "${project.properties["javac.optimize"]}" == "on"
```
Comment (lewismc) [inline build.gradle.kts]: Yes please remove. See my comment on other PR and coordinate with @AzureTriple

**Review by lewismc — CHANGES_REQUESTED** [review]
See minor comment @jbsimmon thank you

**Comment by lewismc** [comment]
Please merge once comment is addressed. Thanks

**Comment by jbsimmon** [comment]
Addressed comments, can merge once someone reviews?

**Pull request merged**